sound/soc/sof/intel/hda-dsp.c

```diff
@@ -364,19 +364,12 @@ static int hda_dsp_wait_d0i3c_done(struct snd_sof_dev *sdev)
 
 static int hda_dsp_send_pm_gate_ipc(struct snd_sof_dev *sdev, u32 flags)
 {
-	struct sof_ipc_pm_gate pm_gate;
-	struct sof_ipc_reply reply;
-
-	memset(&pm_gate, 0, sizeof(pm_gate));
+	const struct sof_ipc_pm_ops *pm_ops = sdev->ipc->ops->pm;
 
-	/* configure pm_gate ipc message */
-	pm_gate.hdr.size = sizeof(pm_gate);
-	pm_gate.hdr.cmd = SOF_IPC_GLB_PM_MSG | SOF_IPC_PM_GATE;
-	pm_gate.flags = flags;
+	if (pm_ops && pm_ops->set_power_state)
+		return pm_ops->set_power_state(sdev, flags);
```

Comment:
the commit description and code don't seem aligned. also set power state and pm_gate are different concepts. This is confusing.

Reply:
let 's use  set_pm_gate, more accurate.

Reply:
use pm_gate